lib/config/genconfig/hadeployconfig_test.go

```diff
@@ -0,0 +1,1425 @@
+	}
+	if upto >= 21 {
+		input(b, "\r")
+	}
+	if upto >= 22 {
+		input(b, "arn-os-snapshot-role"+"\r")
+	}
+	if upto >= 23 {
+		input(b, "ASKEIDSKFLDKDFJERKDS"+"\r")
+	}
+	if upto >= 24 {
+		input(b, "asdfelkjsldfklsdjflsdfjdklsfjsdfsldkfjsd"+"\r")
+	}
+	if upto >= 25 {
+		input(b, "pg.automate.com:5432"+"\r")
+	}
+	if upto >= 26 {
+		input(b, "root"+"\r")
+	}
+	if upto >= 27 {
+		input(b, "root1234"+"\r")
+	}
+	if upto >= 28 {
+		input(b, "dbUser"+"\r")
+	}
+	if upto >= 29 {
+		input(b, "dbUser1234"+"\r")
+	}
+	if upto >= 30 {
+		input(b, "\r")
+	}
+	if upto >= 31 {
+		input(b, "\r")
+	}
+	if upto >= 32 {
+		input(b, "\r")
+	}
+	if upto >= 33 {
+		input(b, "automate-backup"+"\r")
+	}
+	if upto >= 34 {
+		input(b, "AKEKFKEIDKETODDFEDFE"+"\r")
+	}
+	if upto >= 35 {
+		input(b, "sdfsdflsdlfkjsdlkjeflilsfldijflsdkfjddfd"+"\r")
+	}
+	if upto >= 36 {
+		input(b, "asia"+moveDown+"\r")
+	}
+}
+
+func TestPromptExternalPostgresqlRootCert(t *testing.T) {
+	b := bytes.NewBuffer([]byte(""))
+	in := io.NopCloser(b)
+
+	input(b, moveDown+"\r")
+
+	certFilePath := "test.pem"
+	certFileContent := "ROOT CA"
+	input(b, certFilePath+"\r")
+
+	mfsu := &fileutils.MockFileSystemUtils{
+		ReadFileFunc: func(filepath string) (data []byte, err error) {
+			if filepath == certFilePath {
+				data = []byte(certFileContent)
+			}
+			return
+		},
+		StatFunc: func(name string) (fi os.FileInfo, err error) {
+			if name == certFilePath {
+				fi = fileutils.MockFileInfo{IsDirFunc: func() bool { return false }}
+			} else {
+				err = errors.New("file not found")
+			}
+			return
+		},
+	}
+	outBuf := &bytes.Buffer{}
+	out := ioutils.NewNopWriteCloser(outBuf)
+	p := pmt.PromptFactory(in, out, mfsu)
+
+	c := HaDeployConfigGen{
+		Prompt:    p,
+		FileUtils: mfsu,
+		Config:    &config.HaDeployConfig{},
+	}
+
+	c.Config.InitExternal().InitDatabase().Type = "aws"
+
+	err := c.PromptExternalPostgresqlRootCert()
+
+	assert.NoError(t, err)
+
+	assert.Equal(t, certFileContent, c.Config.External.Database.PostgreSQL.PostgresqlRootCert)
+}
+
+func TestPromptExternalOpenSearchRootCert(t *testing.T) {
+	b := bytes.NewBuffer([]byte(""))
+	in := io.NopCloser(b)
+
+	input(b, moveDown+"\r")
+
+	certFilePath := "test.pem"
+	certFileContent := "ROOT CA"
+	input(b, certFilePath+"\r")
+
+	mfsu := &fileutils.MockFileSystemUtils{
+		ReadFileFunc: func(filepath string) (data []byte, err error) {
+			if filepath == certFilePath {
+				data = []byte(certFileContent)
+			}
+			return
+		},
+		StatFunc: func(name string) (fi os.FileInfo, err error) {
+			if name == certFilePath {
+				fi = fileutils.MockFileInfo{IsDirFunc: func() bool { return false }}
+			} else {
+				err = errors.New("file not found")
+			}
+			return
+		},
+	}
+	outBuf := &bytes.Buffer{}
+	out := ioutils.NewNopWriteCloser(outBuf)
+	p := pmt.PromptFactory(in, out, mfsu)
+
+	c := HaDeployConfigGen{
+		Prompt:    p,
+		FileUtils: mfsu,
+		Config:    &config.HaDeployConfig{},
+	}
+
+	c.Config.InitExternal().InitDatabase().Type = "aws"
```

Comment:
we ca use the consts TYPE_AWS and TYPE_SELF_MANAGED

Reply:
These constants exist in `components/automate-cli/cmd/chef-automate/ha_node_add_onprem.go`
and this is `lib/config/genconfig` this cannot depend on that, it will cause cross dependency.